horovod/mxnet/partition_trainer/pos_trainer.py

```diff
@@ -0,0 +1,174 @@
+from horovod.mxnet.mpi_ops import allreduce, allreduce_
+from horovod.mxnet.mpi_ops import broadcast, broadcast_
+from horovod.mxnet.mpi_ops import init, shutdown
+from horovod.mxnet.mpi_ops import size, local_size,  rank, local_rank
+
+import mxnet as mx
+from collections import OrderedDict, defaultdict
+import types
+import time
+import warnings
+from mxnet.gluon.parameter import Parameter
+
+
+class POS_Trainer(mx.gluon.Trainer):

```

Comment:
Can the name of this be more descriptive? Perhaps something like `PartitionedParameterTrainer` or similar?

Alternatively, do you think this parameter splitting code can be made a feature of the existing `hvd.DistributedTrainer` instead of creating a completely separate implementation?